### ISSUE TITLE
Bypass limit of 1000

### ISSUE THREAD

**Issue by riaqn** [aliparlakci/bulk-downloader-for-reddit#570]
- [x] I am requesting a feature.
- [x] I am running the latest version of BDfR
- [x] I have read the [Opening an issue](../../README.md#configuration)
  
## Description
I understand that it's explicitly said in the document that we cannot bypass the limit of roughly 1000 submissions. But I would really appreciate it if someone can explain to me whether this limit can be workaround-ed by some hack?

**Comment by Serene-Arc** [comment]
No, it can't (except in a few instances). The 1000 post limit is actually from Reddit itself and has to do with their backend system. No matter what you do with the Reddit site, it won't return more than 1000 submissions. Not by search, not your saved list, nothing. The library we use, `praw`, is bound with the same limitations. There is no way for us to get more than 1000 submissions because Reddit just won't give us that data.

There are projects to get around this, like Pushshift, but I'm not entirely sure on how they do that and we can't use it because it would require a completely different backend to use through the code. I think it involves continually scraping Reddit to rebuild the database but obviously we can't do that.

There's also the GPDR in some instances. If you want to get all of your saved items but have more than 1000, then an information request through the GPDR will show all of your saved items.

tldr: Reddit has the information but won't give it to us, there's no way around this through code.

**Comment by riaqn** [comment]
Thank you for the explanation, especially for pointing me to pushshift. I'm reading their doc:
https://github.com/pushshift/api
and it seems that they actually provide API, so the end users don't need to maintain a local database at all. Is there any possibility of adding this backend to  BDFR?

**Comment by Serene-Arc** [comment]
@riaqn no, there isn't a chance of adding it even if it was working. It would require rewriting the core of the program from scratch. Basically, it'd be a completely different project with only a little bit of overlap with the BDFR. It'd be the BDFP essentially.
